mkosi/distributions/serpentos.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import re
+import shutil
+import textwrap
+from collections.abc import Sequence
+
+from mkosi.config import Architecture
+from mkosi.context import Context
+from mkosi.distributions import Distribution, DistributionInstaller, PackageType
+from mkosi.installer.moss import invoke_moss
+from mkosi.log import die
+
+
+class Installer(DistributionInstaller):
+    @classmethod
+    def pretty_name(cls) -> str:
+        return "Serpent OS"
+
+    @classmethod
+    def filesystem(cls) -> str:
+        return "btrfs"
+
+    @classmethod
+    def package_type(cls) -> PackageType:
+        return PackageType.moss
+
+    @classmethod
+    def default_release(cls) -> str:
+        return "rolling"
+
+    @classmethod
+    def default_tools_tree_distribution(cls) -> Distribution:
+        return Distribution.serpentos
+
+    @classmethod
+    def _move_kernel_and_initrd(cls, context: Context) -> None:
+        regexp = re.compile("com\.serpentos\.([a-z][a-z0-9]+)\.(.*)")
```

Comment:
This isn't actually doing what it seems to be doing.
The one layer of escaping is "consumed" by python itself, so the actual regexp string that is used is
`com.serpentos.([a-z][a-z0-9]+).(.*)`. Of course it works, because `.` also matches `.`.
A good general rule is to always use r-strings for regexps.

Python `re` does regexp caching internally. So explicit calls to `re.compile` are generally not needed.